test/jdk/java/awt/Window/MultiWindowApp/ChildAlwaysOnTopTest.java

```diff
@@ -21,142 +21,240 @@
+                public void run() {
+                    win1 = parent == null ? new JDialog() : new JDialog(parent);
+                    win1.setName("Top");
+
+                    win2 = parent == null ? new JDialog() : new JDialog(parent);
+                    win2.setName("Behind");
+
+                    JLabel label = new JLabel("TOP WINDOW");
+                    // top window - green and smaller
+                    win1.setSize(200, 200);
+                    Panel panel = new Panel();
+                    panel.setBackground(Color.GREEN);
+                    panel.add(label);
+                    win1.add(panel);
+                    win1.setAlwaysOnTop(true);
+
+                    // behind window - red and bigger
+                    label = new JLabel("BEHIND WINDOW");
+                    win2.setSize(300, 300);
+                    panel = new Panel();
+                    panel.setBackground(Color.RED);
+                    panel.add(label);
+                    win2.add(panel);
 
-        Robot robot = new Robot();
-        robot.delay(500);
-        robot.waitForIdle();
+                    win1.setVisible(true);
+                    win2.setVisible(true);
+                }
+            });
+
+            robot = new Robot();
+            robot.setAutoDelay(300);
+            robot.waitForIdle();
 
-        SwingUtilities.invokeAndWait(new Runnable() {
-            @Override
-            public void run() {
+            // Scenario 1: Trying to unset the alwaysOnTop (green window)
+            // by setting the setVisible to true for behind (red) window
+            System.out.println(" >> Testing Scenario 1 ...");
+            SwingUtilities.invokeAndWait(()-> {
                 point = win1.getLocationOnScreen();
-                win2.setBounds(win1.getBounds());
                 win2.setVisible(true);
-            }
-        });
+            });
 
-        robot.delay(500);
-        robot.waitForIdle();
+            robot.delay(300);
+            robot.waitForIdle();
+            Color color = robot.getPixelColor(point.x + 100, point.y + 100);
 
-        Color color = robot.getPixelColor(point.x + 100, point.y + 100);
-        if(!color.equals(Color.GREEN)) {
-            win1.dispose();
-            win2.dispose();
-            throw new RuntimeException("alawaysOnTop window is sent back by " +
-                    "another child window setVisible(). " + color);
-        }
+            if (!color.equals(Color.GREEN)) {
+                SwingUtilities.invokeAndWait(()-> {
+                    System.out.println("Green Window Active?: "+ win1.isActive());
+                    System.out.println("Red Window Active? "+ win2.isActive());
+                });
+                saveScreenCapture(caseNo , 1);
+                throw new RuntimeException("Scenario 1: alwaysOnTop window is "+
+                        "sent back by another child window with setVisible(). "
+                        + color);
```

Comment:
I see that we are doing a number of test scenarios but test fails immediately when first error occurs. Can we continue testing, save all screenshots and remember all the failures and then fail at the end if there were errors listing all the errors in all the LaF's that were happened during the run?

Reply:
@azuev-java Thank you for the feedback. I can definitely add this functionality to the test for better failure analysis.

Reply:
I think that's a reasonable suggestion in general to bear in mind when writing tests.
I've seen tests which bail at the first failure and you wish you knew how the rest of it worked out ..
Like (say) a test that iterates over the 4 available L&Fs might be a case.

Reply:
Updated test case to continue testing all scenarios and to throw Exception at the end (in case test fails in any of the scenarios)